src/services/node-checker.ts

```diff
@@ -0,0 +1,376 @@
+import get from 'lodash/get'
+import { Pocket, Configuration, Node, RelayResponse, PocketAAT, HTTPMethod, RpcError } from '@pokt-network/pocket-js'
+import { blockHexToDecimal } from '../utils/block'
+import { checkEnforcementJSON } from '../utils/enforcements'
+
+export type Check = 'sync-check' | 'chain-check' | 'archival-check'
+
+export type NodeCheckResponse<T> = {
+  node: Node
+  check: Check
+  success: boolean
+  response: string | Error
+  output?: T
+}
+
+export type ChainCheck = {
+  chainID: number
+}
+
+export type SyncCheck = {
+  blockHeight: number
+}
+
+export type ArchivalCheck = {
+  message: string
+}
+
+type ProcessCheck = {
+  success: boolean
+  relayResponse: RelayResponse | Error
+  output: string | number
+}
+
+export class NodeChecker {
+  pocket: Pocket
+  configuration: Configuration | undefined
+
+  constructor(pocket: Pocket, configuration?: Configuration) {
+    this.pocket = pocket
+    this.configuration = configuration || pocket.configuration
+  }
+
+  /**
+   * Returns a decimal representation from a block's hexadecimal response
+   * @param payload object containing the hexadecimal value
+   * @param key key for which the value is stored
+   * @returns decimal representation of the object's response
+   */
+  static parseBlockFromPayload(payload: object, key: string): number {
+    const rawHeight = get(payload, key) || '0'
+
+    return blockHexToDecimal(rawHeight)
+  }
+
+  /**
+   * Perfoms chain check of a node, making sure the chain a node reports to support is the one that is being requested
+   * within the session.
+   * @param node Node to check.
+   * @param data Payload to send to the blockchain, expected to return the blockchain's chainID.
+   * @param blockchainID Blockchain to request data from.
+   * @param aat Pocket Authentication Token object.
+   * @param chainID  blockchain chain's ID to evaluate against.
+   * @param path  optional. Blockchain's path to send the request to.
+   * @returns Response object containing the relay response, request output and boolean
+   * assuring whether the node supports the correct chain or not
+   */
+  async chain(
+    node: Node,
+    data: string,
+    blockchainID: string,
+    aat: PocketAAT,
+    chainID: number,
+    path?: string
+  ): Promise<NodeCheckResponse<ChainCheck>> {
+    let nodeChainID: number
+
+    const isCorrectChain = (payload: object, chainIDArg) => {
+      nodeChainID = NodeChecker.parseBlockFromPayload(payload, 'result')
+      return nodeChainID === chainIDArg
```

Comment:
Any reason this function needs to be an arrow function, and be inside another function?

Reply:
I use arrow functions unless I explicitly need the previous `this` context, helps from my pov reduce unexpected side effects, also is inside the function as a first-class object (variable on this case) to be passed as a param to `processCheck`

Reply:
The thing is, the fact that you're using an arrow function instead a function declaration means that you're _adding_, not removing, unexpected side-effects, as arrow functions arrived later than function declarations :). As we don't need the `this`, it's a matter of style, but I personally dislike defining utility functions like these inside a method, when they perfectly could go in a `checker-utilts.ts` file or just defined at the top of the file.